Content: IBM Cloud Starter App removed (#568)
flowforge/website: src/blog/2023/03/ibmcloud-starter-removed.md

---
title: IBM Cloud removes Node-RED starter application
subtitle: Get up and running with a Starter Application for Node-RED with FlowForge 

> description: IBM Cloud has recently announced that theyw ill no longer be providing their Cloud App Service Starter Kits, including the Node-RED Starter Application. Don't worry though, FlowForge has you covered.

Reviewer: ```suggestion
description: IBM Cloud has recently announced that they will no longer be providing their Cloud App Service Starter Kits, including the Node-RED Starter Application. Don't worry though, FlowForge has you covered.
```

> date: 2023-03-29

Reviewer: Neat trick: Feel free to not use:

```suggestion
date: git created
```

Reviewer 2: heh - I've ust suggested the opposite given 11ty's docs advise its a compute expensive way of doing it - and it seems needless to require the website to recompute this static value every single time we deploy.

You can install the relevant nodes in one of two places:

1. **Node-RED Palette Manager:** Click "Menu > Manage Palette > Install". The menu is available via the icon in the top-right of your running Node-RED Instance)

![Screenshot of Node-RED's Manage Palette menu](./images/nr-manage-palette-cloudant.png "Screenshot of Node-RED's Manage Palette menu")

2. **FlowForge Instance Settings:** For a given Instance in FlowForge, click "Settings > Palette". You can then define the npm module name and versions explicitely in the "Installed Modules" section

![Screenshot of FlowForge's "Installed Modules" option in Instance > Settings > Palette](./images/ff-installed-modules.png "Screenshot of FlowForge's 'Installed Modules' option in Instance > Settings > Palette")

It's also easy to setup [Environment Variables](/blog/2023/01/environment-variables-in-node-red/) for Node-RED in FlowForge for when you integrate with external services like APIs too.

## Security

> As with IBM Cloud, FlowForge makes it very easy to secure your Node-RED Applications. FlowForge offers three tiers of security options on your Node-RED Instances to secure any exposed HTTP routes on your Node-RED instance, e.g. REST API endpoints or Node-RED Dashboard.

Reviewer: ```suggestion
As with IBM Cloud, FlowForge makes it very easy to secure your Node-RED Applications. [FlowForge offers three tiers of security](https://flowforge.com/docs/user/instance-settings/#security) options on your Node-RED Instances to secure any exposed HTTP routes on your Node-RED instance, e.g. REST API endpoints or Node-RED Dashboard.
```

- **FlowForge Credentials**: Visitors can use their FlowForge username/password in order to access the endpoints. This also includes SSO if you have that configured for your FlowFoerge Team.

## Migrating Existing Instances

If you're looking to move your Node-RED applications from IBM Cloud, then you can do so through one of two options:

### Node-RED Tools Plugin

You can use our Node-RED Tools plugin to migrate your flows and credentials over to FlowForge. You can read the details in our [Migration Guide](https://flowforge.com/docs/user/migration/), which also includes instructions on how to export your environment variables too.

### Manual Import

This will only enable you to import your Flows, not the associated credentials.

1. Export your existing `flows.json` from your IBM Cloud-hosted Node-RED instance by choosing "Export > All Flows > Download" within Node-RED
2. Create a new Application on FlowForge
3. Once created, click the Node-RED instance that has been generated within your Application
4. Click "Settings"
5. Scroll down and select the "Import Instance" option

> 6. Choose your `flows.json` file that you downloaded earlier.

Reviewer: You're missing the environment variables? Or doesn't IBM Cloud use those?

Author: If you're doing this manually, then no easy way of exporting from IBM Cloud that we are aware of (without being able to go in and experiment)

---

> title: IBM Cloud removes Node-RED starter application

Reviewer 3: fwiw I think this is a stronger title and will get more clicks from social media.

> title: IBM Cloud removes Node-RED starter application

Author: Not convinced personally, but if we do change it, then we should change the subtitle with it to something like "IBM Cloud has removed their Node-RED starter application - get up and running on FlowForge instead"

> title: IBM Cloud removes Node-RED starter application

Author: If we don't change the title, then I'd be keen to include the migration phrase into the subtitle instead. Not sure the CTR would be hire, given that most people won't know the start application has been removed in the community, so a standalone migration article (as suggested by title) wouldn't engage many people as they wouldn't know _why_ they would need to migrate
